datafusion/optimizer/src/optimizer.rs

```diff
@@ -274,20 +276,98 @@ impl Optimizer {
     pub fn with_rules(rules: Vec<Arc<dyn OptimizerRule + Send + Sync>>) -> Self {
         Self { rules }
     }
+}
+
+struct Mutator<'a> {
```

Comment:
Here is an example of using the `Mutator` API to rewrite `LogicalPlan`s in place